### PR TITLE
String enum support & command-line args for using references

### PR DESCRIPTION
**Pull request by mprudhom**

### CODE REVIEW

File: dist/typescript-json-schema.js [hunk 1]
```diff
@@ -1,262 +1,350 @@
...
-})(TJS = exports.TJS || (exports.TJS = {}));
-if (typeof window === "undefined" && require.main === module) {
-    if (process.argv[3]) {
-        TJS.exec(process.argv[2], process.argv[3]);
-    }
-    else {
-        console.log("Usage: node typescript-json-schema.js <path-to-typescript-files> <type>\n");
-    }
-}
+var ts = require("typescript");
```
Comment (domoritz) [inline dist/typescript-json-schema.js]: What changed about this file? Is it line endings?

Comment (mprudhom) [inline dist/typescript-json-schema.js]: I suspect it was just line endings, although I'm surprised github doesn't filter that out of the diff.

**Comment by YousefED** [comment]
Great stuff guys, thanks! Vega also looks interesting.

Let me know whether or not you're still working on this (e.g.: adding yargs) - for now I'll hold off the merge

File: dist/typescript-json-schema.js (hunk 1 above)
Comment (domoritz) [inline dist/typescript-json-schema.js]: It's hard to see what happened. I believe there is some git option to ignore line endings. Maybe this helps https://help.github.com/articles/dealing-with-line-endings/

**Comment by mprudhom** [comment]
I added `yargs` parsing support in bin/typescript-json-schema.

I think I'm done for now; feel free to merge if everything looks OK to you.

**Comment by YousefED** [comment]
Thanks!

**Pull request merged**